Sources/App/Logic/WorkspacePath+Logic.swift

```diff
@@ -36,9 +36,39 @@ extension Logic.WorkspacePath {
       throw MockaError.workspacePathNotFolder
     }
   }
+
+  /// Checks if a folder exists in the given path.
+  /// Creates the folder if it does not exist.
+  /// - Parameter path: The path in wich we want the folder.
+  static func checkPathAndCreateFolderIfNeeded(_ path: String) throws {
+    do {
+      try Self.isFolder(URL(fileURLWithPath: path))
+    } catch MockaError.workspacePathDoesNotExist {
+      // Try to create the folder if it does not exist.
+      do {
+        try Self.createFolder(path)
+      } catch {
+        throw error
+      }
+
+    } catch {
+      throw error
+    }
+  }
```

Comment:
You should understand what errors can be thrown and make corresponding `MockaError` entries

Reply:
I improved `checkURLAndCreateFolderIfNeeded(at url: URL?)` with specified thrown errors. I also removed `isFolder()` function because not used anymore.